src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -670,6 +681,38 @@ install_packages_in_root (RpmOstreeTreeComposeContext  *self,
   return TRUE;
 }
 
+static gboolean
+parse_treefile_to_json (const char    *treefile_path,
+                        JsonParser   **out_parser,
+                        GError       **error)
+{
+#ifdef HAVE_RUST
+  g_autofree char *fdpath = NULL;
+  g_auto(GLnxTmpfile) json_contents = { 0, };
+  if (opt_yaml_manifest ||
+      g_str_has_suffix (treefile_path, ".yaml") ||
+      g_str_has_suffix (treefile_path, ".yml"))
+    {
+      if (!glnx_open_anonymous_tmpfile (O_RDWR | O_CLOEXEC, &json_contents, error))
+        return FALSE;
+
+      if (!treefile_read (treefile_path, json_contents.fd))
+        return glnx_throw (error, "Failed to load YAML treefile %s", treefile_path);
+
+      /* or just lseek back to 0 and use json_parser_load_from_data here? */
+      treefile_path = fdpath = g_strdup_printf ("/proc/self/fd/%d", json_contents.fd);
```

Comment:
That won't correctly reference external files like the `passwd` and `treecompose-post.sh` bits.  This is why the current assembler code does the tmpdir/copy dance.

(Shades of `Dockerfile`+`docker build`+"contextdir" here)

Reply:
It still works because as far as the rest of the code is concerned, the treefile path is still the JSON one. We're only passing the `/proc/self` path to `json-glib` when parsing.